TBG/DebugUtils.cpp

```diff
@@ -1,8 +1,25 @@
 #include "DebugUtils.h"
 #include <iostream>
 
-namespace DebugUtils {
-	void Log(std::string message) {
-		std::cout << message << std::endl;
+namespace DebugUtils
+{
+	void Log(const std::string &category, Severity severity, const std::string &message)
+	{
+		switch (severity)
+		{
+		case Info:
+			std::cout << "[" << category << "] "
+					  << "[INFO] ";
+			break;
+		case Warning:
+			std::cout << YELLOW << "[" << category << "] "
+					  << "[WARNING] ";
+			break;
+		case Error:
+			std::cout << RED << "[" << category << "] "
+					  << "[ERROR] ";
+			break;
+		}
+		std::cout << message << RESET << std::endl;
```

Comment:
we can dry this code up. we can start with an empty string that we can concatenate the colors and categories to, then print at the end.